kernel/kernel-api/src/main/java/io/delta/kernel/data/Row.java

```diff
@@ -112,11 +110,11 @@ public interface Row {
      * Return array value of the column located at the given ordinal.
      * Throws error if the column at given ordinal is not of array type,
      */
-    <T> List<T> getArray(int ordinal);
+    ArrayValue getArray(int ordinal);
```

Comment:
Would `getArrayValue` and `getMapValue` be better APIs? 

Since `getMap` makes me think it will return a `java.util.Map`

Reply:
I'm not sure I necessarily agree with
> getMap makes me think it will return a java.util.Map

But I'm relatively indifferent about changing this if everyone is in agreement. Noting that Spark's method is `ColumnVector::getMap --> ColumnarMap`.

Reply:
Let's followup on this when we do the API review